src/libs/actions/PriorityMode.ts

```diff
@@ -101,7 +101,7 @@ function resetHasReadRequiredDataFromStorage() {
 }
 
 function checkRequiredData() {
-    if (allReports === undefined || hasTriedFocusMode === undefined || isInFocusMode === undefined || isLoadingReportData) {
+    if (allReports === undefined || typeof hasTriedFocusMode !== 'boolean' || isInFocusMode === undefined || isLoadingReportData) {
```

Comment:
**Problem that blocks this PR:**
For a new user that has never been switched automatically to **focus** mode, BE never returns `tryFocusMode` after sign-in, so `hasTriedFocusMode` value will always be `null`.
https://github.com/Expensify/App/blob/8f73790b4716ab66aef46f12066910de0a56e0e0/src/libs/actions/PriorityMode.ts#L83-L87

So, the `typeof hasTriedFocusMode !== 'boolean'` condition above will always be true and it's impossible for the auto switch logic to work.

**Solution:**
We expect the BE return either false or true for `tryFocusMode` to indicate whether the user has tried focus mode or not.

Reply:
Bump @techievivek

Reply:
Still valid. No `tryFocusMode` found in OpenApp

<details>
<summaryOpenApp Response</summary>

```
[
    {
        "onyxMethod": "merge",
        "key": "user",
        "value": {
            "isFromPublicDomain": true
        }
    },
    {
        "onyxMethod": "merge",
        "key": "currencyList",
        "value": {
            "AED": {
                "symbol": "Dhs",
                "name": "UAE Dirham",
                "ISO4217": "784"
            },
            "AFN": {
                "symbol": "Af",
                "name": "Afghan Afghani",
                "decimals": 0,
                "ISO4217": "971"
            },
            "ALL": {
                "symbol": "ALL",
                "name": "Albanian Lek",
                "decimals": 0,
                "ISO4217": "008"
            },
            "AMD": {
                "symbol": "դր",
                "name": "Armenian Dram",
                "ISO4217": "051"
            },
            "ANG": {
                "symbol": "NAƒ",
                "name": "Neth Antilles Guilder",
                "ISO4217": "532"
            },
            "AOA": {
                "symbol": "Kz",
                "name": "Angolan Kwanza",
                "ISO4217": "973"
            },
            "ARS": {
                "symbol": "AR$",
                "name": "Argentine Peso",
                "ISO4217": "032"
            },
            "AUD": {
                "symbol": "A$",
                "name": "Australian Dollar",
                "ISO4217": "036"
            },
            "AWG": {
                "symbol": "ƒ",
                "name": "Aruba Florin",
                "ISO4217": "533"
            },
            "AZN": {
                "symbol": "man",
                "name": "Azerbaijani Manat",
                "ISO4217": "944"
            },
            "BAM": {
                "symbol": "KM",
                "name": "Bosnia And Herzegovina Convertible Mark",
                "ISO4217": "977"
            },
            "BBD": {
                "symbol": "Bds$",
                "name": "Barbados Dollar",
                "ISO4217": "052"
            },
            "BDT": {
                "symbol": "Tk",
                "name": "Bangladesh Taka",
                "ISO4217": "050"
            },
            "BGN": {
                "symbol": "лв",
                "name": "Bulgarian Lev",
                "ISO4217": "975"
            },
            "BHD": {
                "symbol": "BHD",
                "name": "Bahraini Dinar",
                "ISO4217": "048"
            },
            "BIF": {
                "symbol": "FBu",
                "name": "Burundi Franc",
                "decimals": 0,
                "ISO4217": "108"
            },
            "BMD": {
                "symbol": "BD$",
                "name": "Bermuda Dollar",
                "ISO4217": "060"
            },
            "BND": {
                "symbol": "BN$",
                "name": "Brunei Dollar",
                "ISO4217": "096"
            },
            "BOB": {
                "symbol": "Bs",
                "name": "Bolivian Boliviano",
                "ISO4217": "068"
            },
            "BRL": {
                "symbol": "R$",
                "name": "Brazilian Real",
                "ISO4217": "986"
            },
            "BSD": {
                "symbol": "BS$",
                "name": "Bahamian Dollar",
                "ISO4217": "044"
            },
            "BTN": {
                "symbol": "Nu.",
                "name": "Bhutan Ngultrum",
                "ISO4217": "064"
            },
            "BWP": {
                "symbol": "P",
                "name": "Botswana Pula",
                "ISO4217": "072"
            },
            "BYN": {
                "symbol": "BR",
                "name": "Belarus Ruble",
                "ISO4217": "933"
            },
            "BYR": {
                "symbol": "BR",
                "name": "Belarus Ruble",
                "decimals": 0,
                "retired": true,
                "retirementDate": "2016-07-01",
                "ISO4217": "974"
            },
            "BZD": {
                "symbol": "BZ$",
                "name": "Belize Dollar",
                "ISO4217": "084"
            },
            "CAD": {
                "symbol": "C$",
                "name": "Canadian Dollar",
                "ISO4217": "124"
            },
            "CDF": {
                "symbol": "CDF",
                "name": "Congolese Franc",
                "ISO4217": "976"
            },
            "CHF": {
                "symbol": "CHF",
                "name": "Swiss Franc",
                "ISO4217": "756"
            },
            "CLP": {
                "symbol": "Ch$",
                "name": "Chilean Peso",
                "decimals": 0,
                "ISO4217": "152"
            },
            "CNY": {
                "symbol": "¥",
                "name": "Chinese Yuan",
                "ISO4217": "156"
            },
            "COP": {
                "symbol": "Col$",
                "name": "Colombian Peso",
                "decimals": 0,
                "ISO4217": "170"
            },
            "CRC": {
                "symbol": "CR₡",
                "name": "Costa Rica Colon",
                "ISO4217": "188"
            },
            "CUC": {
                "symbol": "CUC",
                "name": "Cuban Convertible Peso",
                "ISO4217": "931"
            },
            "CUP": {
                "symbol": "$MN",
                "name": "Cuban Peso",
                "ISO4217": "192"
            },
            "CVE": {
                "symbol": "Esc",
                "name": "Cape Verde Escudo",
                "ISO4217": "132"
            },
            "CZK": {
                "symbol": "Kč",
                "name": "Czech Koruna",
                "ISO4217": "203"
            },
            "DJF": {
                "symbol": "Fdj",
                "name": "Dijibouti Franc",
                "decimals": 0,
                "ISO4217": "262"
            },
            "DKK": {
                "symbol": "Dkr",
                "name": "Danish Krone",
                "ISO4217": "208"
            },
            "DOP": {
                "symbol": "RD$",
                "name": "Dominican Peso",
                "ISO4217": "214"
            },
            "DZD": {
                "symbol": "DZD",
                "name": "Algerian Dinar",
                "ISO4217": "012"
            },
            "EEK": {
                "symbol": "KR",
                "name": "Estonian Kroon",
                "ISO4217": "",
                "retired": true
            },
            "EGP": {
                "symbol": "EGP",
                "name": "Egyptian Pound",
                "ISO4217": "818"
            },
            "ERN": {
                "symbol": "Nfk",
                "name": "Eritrea Nakfa",
                "ISO4217": "232"
            },
            "ETB": {
                "symbol": "Br",
                "name": "Ethiopian Birr",
                "ISO4217": "230"
            },
            "EUR": {
                "symbol": "€",
                "name": "Euro",
                "ISO4217": "978"
            },
            "FJD": {
                "symbol": "FJ$",
                "name": "Fiji Dollar",
                "ISO4217": "242"
            },
            "FKP": {
                "symbol": "FK£",
                "name": "Falkland Islands Pound",
                "ISO4217": "238"
            },
            "GBP": {
                "symbol": "£",
                "name": "British Pound",
                "ISO4217": "826"
            },
            "GEL": {
                "symbol": "ლ",
                "name": "Georgian Lari",
                "ISO4217": "981"
            },
            "GHS": {
                "symbol": "₵",
                "name": "Ghanaian Cedi",
                "ISO4217": "936"
            },
            "GIP": {
                "symbol": "£G",
                "name": "Gibraltar Pound",
                "ISO4217": "292"
            },
            "GMD": {
                "symbol": "D",
                "name": "Gambian Dalasi",
                "ISO4217": "270"
            },
            "GNF": {
                "symbol": "FG",
                "name": "Guinea Franc",
                "decimals": 0,
                "ISO4217": "324"
            },
            "GTQ": {
                "symbol": "Q",
                "name": "Guatemala Quetzal",
                "ISO4217": "320"
            },
            "GYD": {
                "symbol": "GY$",
                "name": "Guyana Dollar",
                "ISO4217": "328"
            },
            "HKD": {
                "symbol": "HK$",
                "name": "Hong Kong Dollar",
                "ISO4217": "344"
            },
            "HNL": {
                "symbol": "HNL",
                "name": "Honduras Lempira",
                "ISO4217": "340"
            },
            "HRK": {
                "symbol": "kn",
                "name": "Croatian Kuna",
                "ISO4217": "191",
                "retired": true
            },
            "HTG": {
                "symbol": "G",
                "name": "Haiti Gourde",
                "ISO4217": "332"
            },
            "HUF": {
                "symbol": "Ft",
                "name": "Hungarian Forint",
                "ISO4217": "348"
            },
            "IDR": {
                "symbol": "Rp",
                "name": "Indonesian Rupiah",
                "ISO4217": "360"
            },
            "ILS": {
                "symbol": "₪",
                "name": "Israeli Shekel",
                "ISO4217": "376"
            },
            "INR": {
                "symbol": "₹",
                "name": "Indian Rupee",
                "ISO4217": "356"
            },
            "IQD": {
                "symbol": "IQD",
                "name": "Iraqi Dinar",
                "decimals": 0,
                "ISO4217": "368"
            },
            "IRR": {
                "symbol": "﷼",
                "name": "Iran Rial",
                "decimals": 0,
                "ISO4217": "364"
            },
            "ISK": {
                "symbol": "kr",
                "name": "Iceland Krona",
                "decimals": 0,
                "ISO4217": "352"
            },
            "JMD": {
                "symbol": "J$",
                "name": "Jamaican Dollar",
                "ISO4217": "388"
            },
            "JOD": {
                "symbol": "JOD",
                "name": "Jordanian Dinar",
                "ISO4217": "400"
            },
            "JPY": {
                "symbol": "¥",
                "name": "Japanese Yen",
                "decimals": 0,
                "ISO4217": "392"
            },
            "KES": {
                "symbol": "KSh",
                "name": "Kenyan Shilling",
                "ISO4217": "404"
            },
            "KGS": {
                "symbol": "KGS",
                "name": "Kyrgyzstani Som",
                "ISO4217": "417"
            },
            "KHR": {
                "symbol": "KHR",
                "name": "Cambodia Riel",
                "ISO4217": "116"
            },
            "KMF": {
                "symbol": "CF",
                "name": "Comoros Franc",
                "decimals": 0,
                "ISO4217": "174"
            },
            "KPW": {
                "symbol": "KP₩",
                "name": "North Korean Won",
                "decimals": 0,
                "ISO4217": "408"
            },
            "KRW": {
                "symbol": "₩",
                "name": "Korean Won",
                "decimals": 0,
                "ISO4217": "410"
            },
            "KWD": {
                "symbol": "KWD",
                "name": "Kuwaiti Dinar",
                "ISO4217": "414"
            },
            "KYD": {
                "symbol": "CI$",
                "name": "Cayman Islands Dollar",
                "ISO4217": "136"
            },
            "KZT": {
                "symbol": "〒",
                "name": "Kazakhstan Tenge",
                "ISO4217": "398"
            },
            "LAK": {
                "symbol": "₭",
                "name": "Lao Kip",
                "decimals": 0,
                "ISO4217": "418"
            },
            "LBP": {
                "symbol": "LBP",
                "name": "Lebanese Pound",
                "decimals": 0,
                "ISO4217": "422"
            },
            "LKR": {
                "symbol": "SL₨",
                "name": "Sri Lanka Rupee",
                "ISO4217": "144"
            },
            "LRD": {
                "symbol": "L$",
                "name": "Liberian Dollar",
                "ISO4217": "430"
            },
            "LSL": {
                "symbol": "M",
                "name": "Lesotho Loti",
                "ISO4217": "426"
            },
            "LTL": {
                "symbol": "Lt",
                "name": "Lithuanian Lita",
                "retirementDate": "2015-08-22",
                "retired": true,
                "ISO4217": "440"
            },
            "LVL": {
                "symbol": "Ls",
                "name": "Latvian Lat",
                "ISO4217": "428",
                "retired": true
            },
            "LYD": {
                "symbol": "LYD",
                "name": "Libyan Dinar",
                "ISO4217": "434"
            },
            "MAD": {
                "symbol": "MAD",
                "name": "Moroccan Dirham",
                "ISO4217": "504"
            },
            "MDL": {
                "symbol": "MDL",
                "name": "Moldovan Leu",
                "ISO4217": "498"
            },
            "MGA": {
                "symbol": "MGA",
                "name": "Malagasy Ariary",
                "decimals": 0,
                "ISO4217": "969"
            },
            "MKD": {
                "symbol": "ден",
                "name": "Macedonian Denar",
                "ISO4217": "807"
            },
            "MMK": {
                "symbol": "Ks",
                "name": "Myanmar Kyat",
                "decimals": 0,
                "ISO4217": "104"
            },
            "MNT": {
                "symbol": "₮",
                "name": "Mongolian Tugrik",
                "ISO4217": "496"
            },
            "MOP": {
                "symbol": "MOP$",
                "name": "Macau Pataca",
                "ISO4217": "446"
            },
            "MRO": {
                "symbol": "UM",
                "name": "Mauritania Ougulya",
                "decimals": 0,
                "retired": true,
                "retirementDate": "2018-07-11",
                "ISO4217": "478"
            },
            "MRU": {
                "symbol": "UM",
                "name": "Mauritania Ougulya",
                "decimals": 0,
                "ISO4217": ""
            },
            "MUR": {
                "symbol": "Rs",
                "name": "Mauritius Rupee",
                "ISO4217": "480"
            },
            "MVR": {
                "symbol": "Rf",
                "name": "Maldives Rufiyaa",
                "ISO4217": "462"
            },
            "MWK": {
                "symbol": "MK",
                "name": "Malawi Kwacha",
                "ISO4217": "454"
            },
            "MXN": {
                "symbol": "Mex$",
                "name": "Mexican Peso",
                "ISO4217": "484"
            },
            "MYR": {
                "symbol": "RM",
                "name": "Malaysian Ringgit",
                "ISO4217": "458"
            },
            "MZN": {
                "symbol": "MTn",
                "name": "Mozambican Metical",
                "ISO4217": "943"
            },
            "NAD": {
                "symbol": "N$",
                "name": "Namibian Dollar",
                "ISO4217": "516"
            },
            "NGN": {
                "symbol": "₦",
                "name": "Nigerian Naira",
                "ISO4217": "566"
            },
            "NIO": {
                "symbol": "NIO",
                "name": "Nicaragua Cordoba",
                "ISO4217": "558"
            },
            "NOK": {
                "symbol": "Nkr",
                "name": "Norwegian Krone",
                "ISO4217": "578"
            },
            "NPR": {
                "symbol": "₨",
                "name": "Nepalese Rupee",
                "ISO4217": "524"
            },
            "NZD": {
                "symbol": "NZ$",
                "name": "New Zealand Dollar",
                "ISO4217": "554"
            },
            "OMR": {
                "symbol": "OMR",
                "name": "Omani Rial",
                "ISO4217": "512"
            },
            "PAB": {
                "symbol": "B",
                "name": "Panama Balboa",
                "ISO4217": "590"
            },
            "PEN": {
                "symbol": "S/.",
                "name": "Peruvian Nuevo Sol",
                "ISO4217": "604"
            },
            "PGK": {
                "symbol": "K",
                "name": "Papua New Guinea Kina",
                "ISO4217": "598"
            },
            "PHP": {
                "symbol": "₱",
                "name": "Philippine Peso",
                "ISO4217": "608"
            },
            "PKR": {
                "symbol": "Rs",
                "name": "Pakistani Rupee",
                "ISO4217": "586"
            },
            "PLN": {
                "symbol": "zł",
                "name": "Polish Zloty",
                "ISO4217": "985"
            },
            "PYG": {
                "symbol": "₲",
                "name": "Paraguayan Guarani",
                "decimals": 0,
                "ISO4217": "600"
            },
            "QAR": {
                "symbol": "QAR",
                "name": "Qatar Rial",
                "ISO4217": "634"
            },
            "RON": {
                "symbol": "RON",
                "name": "Romanian New Leu",
                "ISO4217": "946"
            },
            "RSD": {
                "symbol": "РСД",
                "name": "Serbian Dinar",
                "ISO4217": "941"
            },
            "RUB": {
                "symbol": "₽",
                "name": "Russian Rouble",
                "ISO4217": "643"
            },
            "RWF": {
                "symbol": "RF",
                "name": "Rwanda Franc",
                "decimals": 0,
                "ISO4217": "646"
            },
            "SAR": {
                "symbol": "SAR",
                "name": "Saudi Arabian Riyal",
                "ISO4217": "682"
            },
            "SBD": {
                "symbol": "SI$",
                "name": "Solomon Islands Dollar",
                "ISO4217": "090"
            },
            "SCR": {
                "symbol": "SR",
                "name": "Seychelles Rupee",
                "ISO4217": "690"
            },
            "SDG": {
                "symbol": "SDG",
                "name": "Sudanese Pound",
                "ISO4217": "938"
            },
            "SEK": {
                "symbol": "Skr",
                "name": "Swedish Krona",
                "ISO4217": "752"
            },
            "SGD": {
                "symbol": "S$",
                "name": "Singapore Dollar",
                "ISO4217": "702"
            },
            "SHP": {
                "symbol": "£S",
                "name": "St Helena Pound",
                "ISO4217": "654"
            },
            "SLL": {
                "symbol": "Le",
                "name": "Sierra Leone Leone",
                "decimals": 0,
                "ISO4217": "694"
            },
            "SOS": {
                "symbol": "So.",
                "name": "Somali Shilling",
                "decimals": 0,
                "ISO4217": "706"
            },
            "SRD": {
                "symbol": "SRD",
                "name": "Surinamese Dollar",
                "ISO4217": "968"
            },
            "STD": {
                "symbol": "Db",
                "name": "Sao Tome Dobra",
                "decimals": 0,
                "retired": true,
                "retirementDate": "2018-07-11",
                "ISO4217": "678"
            },
            "STN": {
                "symbol": "Db",
                "name": "Sao Tome Dobra",
                "ISO4217": ""
            },
            "SVC": {
                "symbol": "SVC",
                "name": "El Salvador Colon",
                "ISO4217": "222"
            },
            "SYP": {
                "symbol": "SYP",
                "name": "Syrian Pound",
                "decimals": 0,
                "ISO4217": "760"
            },
            "SZL": {
                "symbol": "E",
                "name": "Swaziland Lilageni",
                "ISO4217": "748"
            },
            "THB": {
                "symbol": "฿",
                "name": "Thai Baht",
                "ISO4217": "764"
            },
            "TJS": {
                "symbol": "TJS",
                "name": "Tajikistani Somoni",
                "ISO4217": "972"
            },
            "TMT": {
                "symbol": "m",
                "name": "Turkmenistani Manat",
                "ISO4217": "934"
            },
            "TND": {
                "symbol": "TND",
                "name": "Tunisian Dinar",
                "ISO4217": "788"
            },
            "TOP": {
                "symbol": "T$",
                "name": "Tonga Pa'ang",
                "ISO4217": "776"
            },
            "TRY": {
                "symbol": "TL",
                "name": "Turkish Lira",
                "ISO4217": "949"
            },
            "TTD": {
                "symbol": "TT$",
                "name": "Trinidad & Tobago Dollar",
                "ISO4217": "780"
            },
            "TWD": {
                "symbol": "NT$",
                "name": "Taiwan Dollar",
                "ISO4217": "901"
            },
            "TZS": {
                "symbol": "TZS",
                "name": "Tanzanian Shilling",
                "ISO4217": "834"
            },
            "UAH": {
                "symbol": "₴",
                "name": "Ukraine Hryvnia",
                "ISO4217": "980"
            },
            "UGX": {
                "symbol": "USh",
                "name": "Ugandan Shilling",
                "decimals": 0,
                "ISO4217": "800"
            },
            "USD": {
                "symbol": "$",
                "name": "United States Dollar",
                "ISO4217": "840"
            },
            "UYU": {
                "symbol": "$U",
                "name": "Uruguayan New Peso",
                "ISO4217": "858"
            },
            "UZS": {
                "symbol": "UZS",
                "name": "Uzbekistani Som",
                "ISO4217": "860"
            },
            "VEB": {
                "symbol": "Bs.",
                "name": "Venezuelan Bolivar",
                "retired": true,
                "retirementDate": "2008-02-01",
                "ISO4217": ""
            },
            "VEF": {
                "symbol": "Bs.F",
                "name": "Venezuelan Bolivar Fuerte",
                "retired": true,
                "retirementDate": "2018-08-20",
                "ISO4217": "937"
            },
            "VES": {
                "symbol": "Bs.S",
                "name": "Venezuelan Bolivar Soberano",
                "ISO4217": "928"
            },
            "VND": {
                "symbol": "₫",
                "name": "Vietnam Dong",
                "decimals": 0,
                "ISO4217": "704"
            },
            "VUV": {
                "symbol": "Vt",
                "name": "Vanuatu Vatu",
                "decimals": 0,
                "ISO4217": "548"
            },
            "WST": {
                "symbol": "WS$",
                "name": "Samoa Tala",
                "ISO4217": "882"
            },
            "XAF": {
                "symbol": "FCFA",
                "name": "CFA Franc (BEAC)",
                "decimals": 0,
                "ISO4217": "950"
            },
            "XCD": {
                "symbol": "EC$",
                "name": "East Caribbean Dollar",
                "ISO4217": "951"
            },
            "XOF": {
                "symbol": "CFA",
                "name": "CFA Franc (BCEAO)",
                "decimals": 0,
                "ISO4217": "952"
            },
            "XPF": {
                "symbol": "XPF",
                "name": "Pacific Franc",
                "decimals": 0,
                "ISO4217": "953"
            },
            "YER": {
                "symbol": "YER",
                "name": "Yemen Riyal",
                "decimals": 0,
                "ISO4217": "886"
            },
            "ZAR": {
                "symbol": "R",
                "name": "South African Rand",
                "ISO4217": "710"
            },
            "ZMK": {
                "symbol": "ZK",
                "name": "Zambian Kwacha",
                "decimals": 0,
                "retired": true,
                "retirementDate": "2013-01-01",
                "ISO4217": "894"
            },
            "ZMW": {
                "symbol": "ZMW",
                "name": "Zambian Kwacha",
                "cacheBurst": 1,
                "ISO4217": "967"
            }
        }
    },
    {
        "key": "nvp_expensify_activePolicyID",
        "onyxMethod": "merge",
        "value": "E6A4091E53CC10D0"
    },
    {
        "key": "nvp_expensify_frequentlyUsedEmojis",
        "onyxMethod": "merge",
        "value": []
    },
    {
        "key": "nvp_expensify_preferredEmojiSkinTone",
        "onyxMethod": "merge",
        "value": "default"
    },
    {
        "key": "nvp_isFirstTimeNewExpensifyUser",
        "onyxMethod": "merge",
        "value": false
    },
    {
        "key": "nvp_preferredLocale",
        "onyxMethod": "merge",
        "value": "en"
    },
    {
        "key": "nvp_private_blockedFromChat",
        "onyxMethod": "merge",
        "value": ""
    },
    {
        "key": "nvp_private_blockedFromConcierge",
        "onyxMethod": "merge",
        "value": []
    },
    {
        "key": "nvp_private_lastPaymentMethod",
        "onyxMethod": "merge",
        "value": []
    },
    {
        "key": "nvp_private_pushNotificationID",
        "onyxMethod": "merge",
        "value": "NsT8nBVtXiWbr8t4"
    },
    {
        "onyxMethod": "merge",
        "key": "nvp_priorityMode",
        "value": "default"
    },
    {
        "onyxMethod": "merge",
        "key": "nvp_lastPaymentMethod",
        "value": []
    },
    {
        "onyxMethod": "merge",
        "key": "nvp_isFirstTimeNewExpensifyUser",
        "value": false
    },
    {
        "onyxMethod": "merge",
        "key": "isFirstTimeNewExpensifyUser",
        "value": false
    },
    {
        "onyxMethod": "merge",
        "key": "preferredLocale",
        "value": "en"
    },
    {
        "onyxMethod": "merge",
        "key": "paypal",
        "value": {}
    },
    {
        "onyxMethod": "merge",
        "key": "preferredEmojiSkinTone",
        "value": "default"
    },
    {
        "onyxMethod": "set",
        "key": "frequentlyUsedEmojis",
        "value": []
    },
    {
        "onyxMethod": "merge",
        "key": "private_blockedFromConcierge",
        "value": []
    },
    {
        "onyxMethod": "merge",
        "key": "private_pushNotificationID",
        "value": "NsT8nBVtXiWbr8t4"
    },
    {
        "onyxMethod": "merge",
        "key": "user",
        "value": {
            "isSubscribedToNewsletter": true,
            "validated": true,
            "isUsingExpensifyCard": false
        }
    },
    {
        "onyxMethod": "set",
        "key": "loginList",
        "value": {
            "bernhard.josephus+0050@gmail.com": {
                "partnerID": 1,
                "partnerName": "expensify.com",
                "partnerUserID": "bernhard.josephus+0050@gmail.com",
                "validatedDate": "2024-04-04 14:08:46"
            }
        }
    },
    {
        "onyxMethod": "merge",
        "key": "personalDetailsList",
        "value": {
            "8392101": {
                "accountID": 8392101,
                "avatar": "https://d2k5nsl2zxldvw.cloudfront.net/images/icons/concierge_2022.png",
                "firstName": "Expensify",
                "lastName": "Concierge",
                "status": null,
                "displayName": "Expensify Concierge",
                "login": "concierge@expensify.com",
                "pronouns": "",
                "timezone": {
                    "automatic": true,
                    "selected": "Europe/Moscow"
                },
                "payPalMeAddress": "",
                "phoneNumber": "",
                "validated": true
            },
            "11665625": {
                "accountID": 11665625,
                "avatar": "https://d2k5nsl2zxldvw.cloudfront.net/images/avatars/default-avatar_18.png",
                "firstName": "",
                "lastName": "Expensify",
                "status": null,
                "displayName": "Expensify",
                "login": "notifications@expensify.com",
                "pronouns": "",
                "timezone": {
                    "automatic": true,
                    "selected": "America/Los_Angeles"
                },
                "payPalMeAddress": "",
                "phoneNumber": "",
                "validated": true
            },
            "16593929": {
                "accountID": 16593929,
                "avatar": "https://d2k5nsl2zxldvw.cloudfront.net/images/avatars/default-avatar_18.png",
                "firstName": "Bernhard",
                "lastName": "Josephus",
                "status": null,
                "displayName": "Bernhard Josephus",
                "login": "bernhard.josephus+0050@gmail.com",
                "pronouns": "",
                "timezone": {
                    "automatic": true,
                    "selected": "America/Los_Angeles"
                },
                "payPalMeAddress": "",
                "phoneNumber": "",
                "validated": true,
                "localCurrencyCode": "IDR"
            }
        }
    },
    {
        "onyxMethod": "set",
        "key": "betas",
        "value": [
            "automaticStatementReporting",
            "pdfGoogleChrome",
            "serverViolations",
            "attendees",
            "automatedTaxExemption",
            "billPay",
            "budget",
            "classicRedirect",
            "collectableDepositAccounts",
            "discrepancy",
            "domainContactBilling",
            "domainTwoFactorAuth",
            "duplicateDetection",
            "emailSuppressionBeta",
            "expensifyCard",
            "expensifyCardQBOReconciliation",
            "expensifyCardRapidIncreaseFraud",
            "expensifyCardUSPMProgram",
            "expensifyOrg",
            "fixViolationPushNotification",
            "freePlanSoftLaunch",
            "gusto",
            "inboxCache",
            "inboxHiddenTasks",
            "indirectIntegrationSetup",
            "intacctVAT",
            "loadPolicyAsync",
            "mapReceipt",
            "mergeAPI",
            "mobileSecureReceipts",
            "monthlySettlement",
            "namesAndAvatars",
            "newPricing",
            "newsletterThree",
            "nextSteps",
            "openFaceHamburger",
            "perDiem",
            "policyRooms",
            "pricingCopyChanges",
            "qboInvoices",
            "realtimeReportComments",
            "s2wAnnouncement",
            "scheduledAutoReporting",
            "secureReceipts",
            "secureReceiptsReports",
            "selfDM",
            "selfServiceHardLaunch",
            "smsSignUp",
            "submitPolicy",
            "swipeToWin",
            "taxForMileage",
            "threads",
            "trackExpense",
            "twoFactorAuth",
            "venmoIntegration",
            "workspaceChangeLogs",
            "zenefitsIntegration"
        ]
    },
    {
        "onyxMethod": "merge",
        "key": "countryCode",
        "value": 62
    },
    {
        "onyxMethod": "merge",
        "key": "country",
        "value": "ID"
    },
    {
        "onyxMethod": "merge",
        "key": "account",
        "value": {
            "requiresTwoFactorAuth": false,
            "guideCalendarLink": null,
            "doesDomainHaveApprovedAccountant": false,
            "activePolicyID": "E6A4091E53CC10D0",
            "dismissedReferralBanners": []
        }
    },
    {
        "onyxMethod": "set",
        "key": "private_blockedFromChat",
        "value": ""
    },
    {
        "onyxMethod": "mergecollection",
        "key": "policyRecentlyUsedCategories_",
        "value": []
    },
    {
        "onyxMethod": "mergecollection",
        "key": "policy_",
        "value": {
            "policy_E6A4091E53CC10D0": {
                "isFromFullPolicy": false,
                "id": "E6A4091E53CC10D0",
                "name": "bernhard.josephus+0050's Expenses",
                "role": "admin",
                "type": "personal",
                "owner": "bernhard.josephus+0050@gmail.com",
                "ownerAccountID": 16593929,
                "outputCurrency": "IDR",
                "avatarURL": "",
                "avatar": "",
                "employeeList": [],
                "isPolicyExpenseChatEnabled": false,
                "lastModified": 1712239737384203,
                "chatReportIDAnnounce": 0,
                "chatReportIDAdmins": 0,
                "areCategoriesEnabled": false,
                "areTagsEnabled": false,
                "areDistanceRatesEnabled": false,
                "areWorkflowsEnabled": false,
                "areConnectionsEnabled": false,
                "tax": {
                    "trackingEnabled": false
                }
            }
        }
    },
    {
        "key": "policy_E6A4091E53CC10D0",
        "onyxMethod": "merge",
        "value": {
            "isFromFullPolicy": false,
            "id": "E6A4091E53CC10D0",
            "name": "bernhard.josephus+0050's Expenses",
            "role": "admin",
            "type": "personal",
            "owner": "bernhard.josephus+0050@gmail.com",
            "ownerAccountID": 16593929,
            "outputCurrency": "IDR",
            "avatarURL": "",
            "avatar": "",
            "employeeList": {
                "bernhard.josephus+0050@gmail.com": {
                    "email": "bernhard.josephus+0050@gmail.com",
                    "forwardsTo": "",
                    "role": "admin",
                    "submitsTo": ""
                }
            },
            "isPolicyExpenseChatEnabled": false,
            "lastModified": 1712239737384203,
            "chatReportIDAnnounce": 0,
            "chatReportIDAdmins": 0,
            "areCategoriesEnabled": false,
            "areTagsEnabled": false,
            "areDistanceRatesEnabled": false,
            "areWorkflowsEnabled": false,
            "areConnectionsEnabled": false,
            "tax": {
                "trackingEnabled": false
            },
            "approvalMode": "OPTIONAL",
            "approver": "",
            "approverAccountID": 0,
            "autoReporting": false,
            "autoReportingFrequency": "immediate",
            "autoReportingOffset": 1,
            "customUnits": {
                "660eb462862a5": {
                    "attributes": {
                        "unit": "mi"
                    },
                    "customUnitID": "660eb462862a5",
                    "defaultCategory": "Car",
                    "enabled": true,
                    "name": "Distance",
                    "rates": {
                        "660eb462862b1": {
                            "currency": "IDR",
                            "customUnitRateID": "660eb462862b1",
                            "enabled": true,
                            "name": "Default Rate",
                            "rate": 67
                        }
                    }
                }
            },
            "defaultBillable": false,
            "description": "",
            "disabledFields": {
                "defaultBillable": true,
                "reimbursable": false
            },
            "fieldList": {
                "expensify_text_title": {
                    "defaultExternalID": null,
                    "defaultValue": "{report:type} {report:startdate}",
                    "deletable": true,
                    "disabledOptions": [],
                    "externalID": null,
                    "externalIDs": [],
                    "fieldID": "text_title",
                    "isTax": false,
                    "keys": [],
                    "name": "title",
                    "orderWeight": 0,
                    "origin": null,
                    "target": "expense",
                    "type": "formula",
                    "value": null,
                    "values": []
                }
            },
            "harvesting": {
                "enabled": false
            },
            "hasMultipleTagLists": false,
            "preventSelfApproval": false,
            "reimburserAccountID": 0,
            "reimburserEmail": null,
            "requiresCategory": false,
            "requiresTag": false,
            "submitsTo": 0
        }
    },
    {
        "key": "policyMembers_E6A4091E53CC10D0",
        "onyxMethod": "set",
        "value": {
            "16593929": {
                "role": "admin"
            }
        }
    },
    {
        "key": "policyCategories_E6A4091E53CC10D0",
        "onyxMethod": "set",
        "value": {
            "Advertising": {
                "enabled": true,
                "name": "Advertising"
            },
            "Benefits": {
                "enabled": true,
                "name": "Benefits"
            },
            "Car": {
                "enabled": true,
                "name": "Car"
            },
            "Equipment": {
                "enabled": true,
                "name": "Equipment"
            },
            "Fees": {
                "enabled": true,
                "name": "Fees"
            },
            "Home Office": {
                "enabled": true,
                "name": "Home Office"
            },
            "Insurance": {
                "enabled": true,
                "name": "Insurance"
            },
            "Interest": {
                "enabled": true,
                "name": "Interest"
            },
            "Labor": {
                "enabled": true,
                "name": "Labor"
            },
            "Maintenance": {
                "enabled": true,
                "name": "Maintenance"
            },
            "Materials": {
                "enabled": true,
                "name": "Materials"
            },
            "Meals and Entertainment": {
                "enabled": true,
                "name": "Meals and Entertainment"
            },
            "Office Supplies": {
                "enabled": true,
                "name": "Office Supplies"
            },
            "Other": {
                "enabled": true,
                "name": "Other"
            },
            "Professional Services": {
                "enabled": true,
                "name": "Professional Services"
            },
            "Rent": {
                "enabled": true,
                "name": "Rent"
            },
            "Taxes": {
                "enabled": true,
                "name": "Taxes"
            },
            "Travel": {
                "enabled": true,
                "name": "Travel"
            },
            "Utilities": {
                "enabled": true,
                "name": "Utilities"
            }
        }
    },
    {
        "key": "policyTags_E6A4091E53CC10D0",
        "onyxMethod": "set",
        "value": {
            "Tag": {
                "name": "Tag",
                "orderWeight": 0,
                "required": false,
                "tags": []
            }
        }
    },
    {
        "onyxMethod": "mergecollection",
        "key": "report_",
        "value": {
            "report_2827813452752048": {
                "reportID": "2827813452752048",
                "reportName": "",
                "type": "chat",
                "chatType": "selfDM",
                "ownerAccountID": 16593929,
                "managerID": 0,
                "policyID": "_FAKE_",
                "participants": [],
                "participantAccountIDs": [],
                "visibleChatMemberAccountIDs": [],
                "isPinned": false,
                "lastReadTime": null,
                "lastMentionedTime": null,
                "lastReadSequenceNumber": 0,
                "lastVisibleActionCreated": "2024-04-04 14:08:34.818",
                "lastVisibleActionLastModified": "2024-04-04 14:08:34.818",
                "lastMessageText": "",
                "lastActorAccountID": 16593929,
                "notificationPreference": "always",
                "stateNum": 0,
                "statusNum": 0,
                "oldPolicyName": "",
                "visibility": null,
                "isOwnPolicyExpenseChat": false,
                "lastMessageHtml": "",
                "iouReportID": null,
                "hasOutstandingChildRequest": false,
                "policyName": "",
                "hasParentAccess": null,
                "parentReportID": null,
                "parentReportActionID": null,
                "writeCapability": "all",
                "description": "",
                "isDeletedParentAction": null,
                "fieldList": null,
                "total": 0,
                "unheldTotal": 0,
                "currency": "USD",
                "submitterPayPalMeAddress": "",
                "chatReportID": null,
                "isWaitingOnBankAccount": false,
                "nonReimbursableTotal": 0,
                "isCancelledIOU": false,
                "avatarUrl": ""
            },
            "report_6701794158921935": {
                "reportID": "6701794158921935",
                "reportName": "Chat Report",
                "type": "chat",
                "chatType": "",
                "ownerAccountID": 0,
                "managerID": 0,
                "policyID": "_FAKE_",
                "participants": {
                    "8392101": {
                        "hidden": false
                    },
                    "16593929": {
                        "hidden": false
                    }
                },
                "participantAccountIDs": [
                    8392101
                ],
                "visibleChatMemberAccountIDs": [
                    8392101
                ],
                "isPinned": true,
                "lastReadTime": "2024-04-04 14:08:34.818",
                "lastMentionedTime": null,
                "lastReadSequenceNumber": 0,
                "lastVisibleActionCreated": "2024-04-04 14:08:35.881",
                "lastVisibleActionLastModified": "2024-04-04 14:08:35.881",
                "lastMessageText": "Welcome to Expensify 👋 Hey there, I'm Concierge! If you have any questions abou…",
                "lastActorAccountID": 8392101,
                "notificationPreference": "always",
                "stateNum": 0,
                "statusNum": 0,
                "oldPolicyName": "",
                "visibility": null,
                "isOwnPolicyExpenseChat": false,
                "lastMessageHtml": "<h1>Welcome to Expensify</h1>👋 Hey there, I'm Concierge! If you have any questions about Expensify, you can always chat with me here 24-7 for fast and reliable support. I'm happy to help!",
                "iouReportID": null,
                "hasOutstandingChildRequest": false,
                "policyName": null,
                "hasParentAccess": null,
                "parentReportID": null,
                "parentReportActionID": null,
                "writeCapability": "all",
                "description": "",
                "isDeletedParentAction": null,
                "fieldList": null,
                "total": 0,
                "unheldTotal": 0,
                "currency": "USD",
                "submitterPayPalMeAddress": "",
                "chatReportID": null,
                "isWaitingOnBankAccount": false,
                "nonReimbursableTotal": 0,
                "isCancelledIOU": false,
                "avatarUrl": ""
            },
            "report_8810944956641220": {
                "reportID": "8810944956641220",
                "reportName": "#public",
                "type": "chat",
                "chatType": "policyRoom",
                "ownerAccountID": 0,
                "managerID": 0,
                "policyID": "08A7DEE4E76566BF",
                "participants": [],
                "participantAccountIDs": [],
                "visibleChatMemberAccountIDs": [],
                "isPinned": false,
                "lastReadTime": "2024-04-04 14:07:03.469",
                "lastMentionedTime": null,
                "lastReadSequenceNumber": 0,
                "lastVisibleActionCreated": "2024-03-13 07:41:37.155",
                "lastVisibleActionLastModified": "2024-03-13 07:41:37.155",
                "lastMessageText": "set the room description to:  a",
                "lastActorAccountID": 12710168,
                "notificationPreference": "daily",
                "stateNum": 0,
                "statusNum": 0,
                "oldPolicyName": "",
                "visibility": "public",
                "isOwnPolicyExpenseChat": false,
                "lastMessageHtml": "<muted-text>set the room description to: <pre>a<br /></pre></muted-text>",
                "iouReportID": null,
                "hasOutstandingChildRequest": false,
                "policyName": "Bernhard.josephus's Workspace 4",
                "hasParentAccess": null,
                "parentReportID": null,
                "parentReportActionID": null,
                "writeCapability": "all",
                "description": "<pre>a<br /></pre>",
                "isDeletedParentAction": null,
                "fieldList": null,
                "total": 0,
                "unheldTotal": 0,
                "currency": "USD",
                "submitterPayPalMeAddress": "",
                "chatReportID": null,
                "isWaitingOnBankAccount": false,
                "nonReimbursableTotal": 0,
                "isCancelledIOU": false,
                "avatarUrl": ""
            }
        }
    },
    {
        "onyxMethod": "set",
        "key": "cardList",
        "value": {}
    },
    {
        "onyxMethod": "merge",
        "key": "userWallet",
        "value": {
            "availableBalance": 0,
            "bankAccountID": 4145921,
            "currentBalance": 0,
            "currentStep": "",
            "hasAcceptedTerms": false,
            "hasFailedOnfido": false,
            "isPendingOnfidoResult": false,
            "remainingWalletLimit": 49999,
            "state": 0,
            "tier": 1,
            "tierName": "SILVER",
            "walletLimit": 50000,
            "walletLimitEnforcementPeriod": 60,
            "walletLinkedAccountID": 0,
            "walletLinkedAccountType": ""
        }
    },
    {
        "onyxMethod": "merge",
        "key": "bankAccountList",
        "value": {}
    },
    {
        "onyxMethod": "mergecollection",
        "key": "transactions_",
        "value": []
    },
    {
        "onyxMethod": "merge",
        "key": "private_personalDetails",
        "value": []
    },
    {
        "onyxMethod": "merge",
        "key": "nvp_quickActionGlobalCreate",
        "value": {}
    }
]
```

</details>

Reply:
Extremely sorry for missing the ping, I am taking a look now.

Reply:
> For a new user that has never been switched automatically to focus mode, BE never returns tryFocusMode after sign-in, so hasTriedFocusMode value will always be null.

So the problem over here is that when a user logs in we don't return the value for `hasTriedFocusMode` and we can't default it to `false` because the user might have already tried the focus mode? Is that correct understanding?

Reply:
> we can't default it to false because the user might have already tried the focus mode?

Yes.

Reply:
Ok, just checked the backend code and realized we are defaulting the value to null rather than false.
```
$hasTriedFocusMode = $nvps[NVP::TRY_FOCUS_MODE] ?? null;
```

Reply:
We can default to false here and that should fix the issue.

Reply:
I think an easier solution would be to just update code to handle null value, no?

```
hasTriedFocusMode = !!val;
```
As you already mentioned, this only happens for users whom we haven't switched to focus mode. So if the value is null, can we default to `false`?

Reply:
@techievivek we can't default it to false in the FE. `checkRequiredData` checks whether `hasTriedFocusMode` is ready or not. When we log out from the app, `hasTriedFocusMode` becomes null because the onyx is cleared. If we default it to false, `checkRequiredData` will think `hasTriedFocusMode` is ready.

> Ok, just checked the backend code and realized we are defaulting the value to null rather than false.

> $hasTriedFocusMode = $nvps[NVP::TRY_FOCUS_MODE] ?? null;

Also, I think the issue might be bigger than this because even after I switch to focus mode from the preferences page, and then relogin, OpenApp still doesn't give me `TRY_FOCUS_MODE` value.

But If I pass `automatic: true` to `UpdateChatPriorityMode`, then OpenApp will give me `TRY_FOCUS_MODE`

Reply:
> we can't default it to false in the FE. checkRequiredData checks whether hasTriedFocusMode is ready or not. When we log out from the app, hasTriedFocusMode becomes null because the onyx is cleared. If we default it to false, checkRequiredData will think hasTriedFocusMode is ready.

Good point. 👍 

> Also, I think the issue might be bigger than this because even after I switch to focus mode from the preferences page, and then relogin, OpenApp still doesn't give me TRY_FOCUS_MODE value.

> But If I pass automatic: true to UpdateChatPriorityMode, then OpenApp will give me TRY_FOCUS_MODE

Yeah, that is expected behavior. We set `TRY_FOCUS_MODE` to `true` in the backend only when the request has come after showing you the modal for it and not when users update their preference manually. I am not sure why it was implemented like that honestly.

Reply:
And we do that with the help of `automatic` param.

Reply:
I can think of 2 improvements here based on the above discussion.
1. Return `false` as the default value from the backend and not null.
2. Set `TRY_FOCUS_MODE` to `true` when users manually update their preference to `#focus` so we don't show them the try #focus mode option.

Does that sound good?

Reply:
Sounds good 👍

Reply:
Great, in that case, we won't need any frontend changes, right?

Reply:
We probably still need it because currently, we check whether `hasTriedFocusMode` is undefined or not, but it's null after signing out.

When we call OpenApp, we set the `isLoadingReportData` optimistically to true and set it to false in `finallyData`.
https://github.com/Expensify/App/blob/f95bf85925686e64bf6c5bcfb6e66722ff6934fc/src/libs/actions/PriorityMode.ts#L93-L96

`checkRequiredData` is called every time `isLoadingReportData` is updated. When OpenApp API completes, it will process the `finallyData` first. (OpenApp is a WRITE request)
https://github.com/Expensify/App/blob/f95bf85925686e64bf6c5bcfb6e66722ff6934fc/src/libs/actions/OnyxUpdates.ts#L30-L51

So, `isLoadingReportData` becomes false first before we have the `hasTriedFocusMode` data which will cause the `checkRequiredData` to resolve the promise.

I think we should retest after the BE change

Reply:
I want to clarify the behavior here. 

When we see the try focus modal, App makes a parallel backend request to update the value for `TRY_FOCUS_MODE` NVP and sets it to `true`, right? Now, if we log out and log back in, we don't see the same modal. So what's the bug in here?

Reply:
@ishpaul777 Can you clarify on the above comments, thanks.

Reply:
> which is fixed here

Hmm, I am not on this branch and still not able to reproduce this bug.

Reply:
> the original bug reported was when user has already tried focus mode and switch to most recent in preferences iand then on every relogin it again auto switch to priorty mode

This doesn't seem to be true, you can see the [code](https://github.com/Expensify/App/blob/d8599ac51d80c8cbd198c07436e73d01109e0b14/src/components/FocusModeNotification.tsx#L18) that it sends a request to update the priority mode and sets the TRY_FOCUS_MODE nvp.

Reply:
Yeah, I can't reproduce it anymore too. When I log the value of `hasTriedFocusMode` after signing out, it returns `undefined` now instead of `null`.

Reply:
I found out why it returns `undefined` now. This happens after we update the key from `tryFocusMode` to `nvp_tryFocusMode`. Onyx detect it as a collection key, and all collection key becomes undefined when the onyx is cleared. If it's not a collection key, it still becomes null.

I think it's a bug. When we clear the onyx, we [set the data to the default if exist or null by default](https://github.com/Expensify/react-native-onyx/blob/667e0d2daa67e06ee02cc1162ce3731c3cac18f2/lib/Onyx.ts#L497) then broadcast it to the onyx subscriber.

If it's a non-collection key, it will just [broadcast the value that we pass to the broadcast function](https://github.com/Expensify/react-native-onyx/blob/667e0d2daa67e06ee02cc1162ce3731c3cac18f2/lib/OnyxUtils.js#L692), which is null.

But if it's a collection key, it will get the cached list of keys within the same collection (nvp_) which is [already dropped](https://github.com/Expensify/react-native-onyx/blob/667e0d2daa67e06ee02cc1162ce3731c3cac18f2/lib/Onyx.ts#L542) (except nvp_preferredLocale, idk why), so we get basically an empty key list and when we try to access the value of `nvp_tryFocusMode`, we get undefined.

Reply:
I think the bug is fixed somehow, I can ask the QA team to retest it.

Reply:
It's no longer valid anymore because of the onyx bug. If we fix the onyx bug, then this issue will happen again.

If you want try to reproduce the issue ignoring the onyx bug,
https://github.com/Expensify/App/blob/2995f2c79751d5096919eb98a26e0bd7b8533658/src/libs/actions/PriorityMode.ts#L77

you can try default the value to null
```
hasTriedFocusMode = val ?? null;
```

Reply:
> cause any bugs in App that surely suggest it is a bug

I don't know 100%, but I think not because it's just a value difference returned between the collection (undefined) and non-collection key (null) when cleared.

Also not sure whether the key is falsely detected as a collection key causes any issue or not. I just want to say if later the value returned is null again, then we will have the issue back. Maybe we can close it for now and reopen if the issue happen again in the future.

Reply:
If there is a bug with Onyx, I think we should bring it to the open-source channel for a detailed discussion since that could potentially impact other areas of our codebase. As for this particular issue, since it's no longer reproducible in our current version of the codebase, we can go ahead and close it for now.